src/wp-admin/includes/image.php

```diff
@@ -646,19 +646,40 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
  *
  * @since 2.5.0
  *
- * @param string $str
- * @return int|float
+ * @param string $str Fraction string.
+ * @return int|float Returns calculated fraction or integer 0 on invalid input.
  */
 function wp_exif_frac2dec( $str ) {
-	if ( false === strpos( $str, '/' ) ) {
-		return $str;
+	if ( ! is_scalar( $str ) || is_bool( $str ) ) {
+		return 0;
+	}
+
+	if ( ! is_string( $str ) ) {
+		return $str; // This can only be an integer or float, so this is fine.
+	}
+
+	// Fractions passed as a string must contain a single `/`.
+	if ( substr_count( $str, '/' ) !== 1 ) {
+		if ( is_numeric( $str ) ) {
+			return (float) $str;
+		}
+
+		return 0;
 	}
 
 	list( $numerator, $denominator ) = explode( '/', $str );
-	if ( ! empty( $denominator ) ) {
-		return $numerator / $denominator;
+
+	// Both the numerator and the denominator must be numbers.
+	if ( ! is_numeric( $numerator ) || ! is_numeric( $denominator ) ) {
+		return 0;
+	}
+
+	// The denominator must not be zero.
+	if ( 0 == $denominator ) { // phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison -- Deliberate loose comparison.
```

Comment:
```suggestion
	// phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison -- Deliberate loose comparison.
	if ( 0 == $denominator ) {
```

`phpcs:ignore` ignores [both the current and next line](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file) so putting it on a line above avoids accidentally ignoring more than is intended.

An expanded comment explaining that this can't be converted to a falsey check due to the possibility of '0.0' would be good too https://3v4l.org/LiF4J

Reply:
@peterwilsoncc The "ignoring the current and next line" only applies to lines which only contain the ignore annotation, not to trailing comments.

The reason I put it as a trailing comment instead of stand-alone is that there is already a comment above the line.
Inserting the ignore annotation between them would divorce the comment from the code it applies to and _might_ tempt someone to then convert the two comment lines to a multi-line `/* */` comments syntax, which would break the ignore annotation.

Expanding the comment wouldn't hurt, but there is a test covering this exact case `'0.0'`, so anyone tempted to change this should be alerted soon enough that that's not a good idea and why due to that specific test failing. (it's the reason I put the test in).